Changed files tree: add context menu and drag&drop for folder nodes

cbjeukendrup opened the pull request:
https://github.com/user-attachments/assets/e6d71923-f5f0-4489-8eca-8591263614c1

Resolves: https://github.com/gitbutlerapp/gitbutler/issues/9182

I could squash the 1st and 2nd commit if preferred. For reviewing, the "Hide whitespace changes" option may be useful.

cbjeukendrup committed:
Add context menu for folder items in changed files tree
Fix typo: succeded -> succeeded

cbjeukendrup committed:
Support dragging folder items in changed files tree

Dragging a folder item is equivalent to selecting the file items contained in that folder and dragging those.

cbjeukendrup committed:
Merge FileContextMenu and FolderContextMenu to avoid code duplication

Byron commented:
This is fantastic, thanks so much! The video was a pleasure to watch.


And it is true, for reviewing in the browser, the hide-whitespace feature is preferable.

<img width="276" height="304" alt="Screenshot 2025-10-30 at 05 25 10" src="https://github.com/user-attachments/assets/39c37b15-c1ef-4a2e-ba9c-d6b892651f89" />

Let's get this reviewed :).

PavelLaptev committed:
Set file icon color in draggable styles

PavelLaptev committed:
Truncate file/folder chip names and improve styling

Adds 'truncate' class to file and folder chip names to prevent overflow. Updates CSS to set overflow: hidden on chip container and prevent icon shrinking for better layout consistency.

PavelLaptev commented:
@cbjeukendrup thanks for the PR! I’ve added a few style fixes.
Overall, it looks good to me.
